### PR TITLE
add organization followers mapping migration

### PR DESCRIPTION
**Pull request by premkiran2**
## Summary
- add a Supabase migration for `organization_followers`
- enforce uniqueness with a composite primary key on `user_id` and `organization_id`
- add indexes to support follower lookup queries
- add RLS policies for reading and managing a user's own follow rows
- add a migration test covering the new schema

## Notes
- `organization_id` is stored as a required `uuid`
- no foreign key was added for `organization_id` because this repo does not currently define an `organizations` table

### CODE REVIEW

**Review by Devayani1612 — CHANGES_REQUESTED** [review]
REQUEST CHANGES - Issue #30: Organization Followers Mapping**
> 
> ### Status:  NOT IMPLEMENTED
> 
> After reviewing all migration files, the core requirements from Issue #30 are **missing**:
> 
> ####  Missing Components:
> - [ ] `organizations` table
> - [ ] `organization_followers` table (many-to-many relationship)
> - [ ] Performance indexes
> - [ ] Row-Level Security policies
> - [ ] Proper migration file
> 

> ####  Critical Issues:
> 
> 1. **No organizations table** - Required parent table for all organization data
> 2. **No organization_followers table** - The main requirement of Issue #30
> 3. **File 4 architecture is incorrect** - Mixing org data into profiles table won't scale
> 
 
> #### Acceptance Criteria Status:
> -  Table created successfully → MISSING
> -  Data integrity maintained → MISSING  
> - Queries return correct followers → MISSING
>

**Comment by premkiran2** [comment]
## Summary
- fix organization search so org accounts appear in the dashboard search bar
- sync missing profile fields from auth metadata into `profiles` on login
- refresh organization search results when the search input is focused
- keep organization search aligned with the app’s actual role values (`Student`, `Organization`)

## Result
- organization accounts become searchable after login/profile sync
- both student and organization users can search for organizations from the home screen
- clicking a search result opens the organization profile page

**Comment by premkiran2** [comment]
I’ve added the Organization table and made the necessary changes suggested by @Devayani1612 .

**Review by Devayani1612 — DISMISSED** [review]
All acceptance criteria from **Issue #30 (CB-303 Database: Organization Followers Mapping)** are successfully implemented:

###  Acceptance Criteria Met:
- ** Table created successfully** - The `organization_followers` table is properly structured with UUID foreign keys and cascading deletes
- ** Data integrity maintained** - Composite primary key `(user_id, organization_id)` prevents duplicate follows; indexes optimize queries
- ** Queries return correct followers** - API functions (`fetchFollowerCount()`, `fetchIsFollowing()`, etc.) correctly handle all follower operations

### Key Implementation Highlights:
1. **Database Design**
   - Composite primary key enforces uniqueness and prevents duplicate follows
   - Cascading deletes maintain referential integrity
   - Strategic indexes on frequently-queried columns for performance

2. **Security**
   - RLS policies correctly restrict writes to authenticated users
   - Users can only manage their own follow records
   - Public read access for discovery features

3. **Code Quality**
   - Modular API functions in `client/src/api/organizations.js`
   - Comprehensive tests covering organizations, followers, and auth sync
   - Well-structured utility functions extracted from App component

File: client/tests/organizations.test.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
+import { beforeEach, describe, expect, it, vi } from "vitest";
```
Comment (Supraja050202) [inline client/tests/organizations.test.js:1]: Good coverage for success and error cases, but missing a test for when Supabase returns null without an error. This edge case should be explicitly tested to ensure proper error handling behavior.

File: client/src/components/PostCard.jsx [hunk 2]
```diff
@@ -1,22 +1,50 @@
...
-      { }
       <div className="flex items-center gap-3 mb-3">
-        <div className="h-10 w-10 rounded-full bg-slate-300 flex items-center justify-center text-sm font-semibold text-slate-700">
-          {post.club_name.charAt(0)}
-        </div>
+        <AvatarBadge src={post.avatarUrl} label={post.club_name} />
         <div>
-          <h3 className="font-semibold text-slate-800">{post.club_name}</h3>
-          <p className="text-xs text-slate-500">Campus Event</p>
+          {canOpenProfile ? (
```
Comment (Supraja050202) [inline client/src/components/PostCard.jsx:31]: post.userId may be undefined for legacy or incomplete posts, which could result in unintended behavior when calling onOpenProfile. Consider adding a guard clause before invoking the callback.

**Review by Supraja050202 — CHANGES_REQUESTED** [review]
This PR demonstrates solid implementation quality with good separation of concerns, readable code structure, and meaningful test coverage. The recent fixes around handling undefined post.userId and adding test coverage for Supabase returning null improve both robustness and edge-case reliability.

The codebase is generally well-organized, and error handling patterns are consistent. However, a few areas still require attention, such as ensuring defensive checks around user-driven callbacks and expanding test coverage for additional Supabase edge cases to fully eliminate silent failure scenarios.

Overall, this is a strong update with improved stability and better test coverage, moving the implementation closer to production readiness.

**Review by Supraja050202 — APPROVED** [review]
Approved

**Review by Jahnavidarisetti — APPROVED** [review]
Good! The organization details is covering all the essential details a user might need from the organization.
